### DISCUSSION TITLE
Trying to relabel a mesh for mixed Poisson

### DISCUSSION THREAD

**Discussion by pzuritas** [firedrakeproject/firedrake#2992] [Firedrake support] [answered]
¡Hi everyone!

I'm new to Firedrake (used to using FEniCS). I've been trying for a while to solve a mixed Poisson problem on a fairly complicated mesh. The mesh is a hollow tetrakaidecahedron with two holes cut out along the x axis (`tkd.msh`). The mesh is provided externally, so I'm relabeling it to define three boundary subdomains for boundary conditions. Probably doing it wrong.

I keep getting `LookupError: BC construction got invalid markers {1}. Valid markers are 'set()'` when trying to define Dirichlet boudary conditions on the relabeled mesh. Anyone got any ideas on how to solve this or anything that might be causing the issue?

I've attached the mesh and the commented MRE: [mre.zip](https://github.com/firedrakeproject/firedrake/files/11775996/mre.zip).

**Comment by ksagiyam** [comment, marked as answer]
`RelabeledMesh` makes a new mesh object with new labels, so you need to make `mark_tkd_boundaries()` function return that new mesh object and use that mesh in the rest of your code.

**Reply by pzuritas** [reply]
That did it, thanks for the clarification!
